modules/post/multi/manage/shell_to_meterpreter.rb

```diff
@@ -138,7 +140,18 @@ def run
       return nil
     end
 
-    payload_name = datastore['PAYLOAD_OVERRIDE'] if datastore['PAYLOAD_OVERRIDE']
+    if datastore['PAYLOAD_OVERRIDE']
+      payload_name = datastore['PAYLOAD_OVERRIDE']
+      payload_info = payload_name.split('/')
+      payload = framework.payloads.create(payload_name).platform.platforms
+
+      if payload_info.first == 'windows'
+        psh_arch = payload.arch
+      else
+        lplat = payload.platform.platforms
+        larch = payload.arch
```

Comment:
I'm not _positive_ this is all the data we need to be set in this case, I'm looking into other cases, but would love feedback on this